corehq/apps/dump_reload/sql/load.py

```diff
@@ -160,11 +160,15 @@ def _group_objects_by_db(objects):
             except KeyError:
                 # in the case of foreign keys the serialized field name is the
                 # name of the foreign key attribute
-                field = [
-                    field for field in model._meta.fields
-                    if field.column == model.partition_attr
-                ][0]
-                partition_value = obj['fields'][field.name]
+                try:
+                    field = [
+                        field for field in model._meta.fields
+                        if field.column == model.partition_attr
+                    ][0]
+                    partition_value = obj['fields'][field.name]
+                except KeyError:
+                    if model.partition_attr == model._meta.pk.attname:
+                        partition_value = obj['pk']
```

Comment:
`partition_value` will not be set if the condition evaluates to false, and I think that would either cause a `NameError` on the next line or use `partition_value` from the previous loop iteration which seems wrong.

Reply:
yea, I guess that should raise an exception

Reply:
8c695c42a7f